concurrent_map.go

```diff
@@ -290,7 +316,27 @@ func (m ConcurrentMap) Keys() []string {
 	return keys
 }
 
-//Reviles ConcurrentMap "private" variables to json marshal.
+// FindOldest search for the oldest key in the key slice.
+func (m ConcurrentMap) FindOldest() string {
+	keys := []string{}
+	for i := 0; i < ShardCount; i++ {
+		shard := m[i]
+		if len(shard.mapKeys) > 0 {
```

Comment:
shard.mapKeys should also be under the lock

Reply:
Done.